### PR TITLE
Add some documentation about generating the doxygen documentation

### LINKED ISSUE #1486: Make doc does not work
#### Problem
`make help` claims that doc should work .Attempting it results in failure.

 #### Proposed Solution
Figure out if we should have a doc step and if yes, then make it actually work

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

People have hard time finding how to generate the documentation.

 #### Summary of Changes
 * Add some documentation in building about generating the documentation.

fixes #1486

### CODE REVIEW

**Review by saurabhst — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Review by jelderton — APPROVED** [review]

File: docs/BUILDING.md [hunk 1]
```diff
@@ -275,6 +275,29 @@ export ANDROID_NDK_HOME=~/Library/Android/sdk/ndk/21.0.6113669
 make -f Makefile-Android
 ```
 
+### Generate the documentation
+
+The doxygen documentation is generated automatically if doxygen is detected on
```
Comment (gerickson) [inline docs/BUILDING.md:280]: doxygen -> Doxygen

File: docs/BUILDING.md [hunk 2]
```diff
@@ -275,6 +275,29 @@ export ANDROID_NDK_HOME=~/Library/Android/sdk/ndk/21.0.6113669
...
+the system. The generated documentation can be found into
+`build/<CONFIG>/docs/html`
```
Comment (gerickson) [inline docs/BUILDING.md:282]: The root of the path will depend on how and where the developer built. Probably best to describe this as "`${top_builddir}/docs/html` where `${top_builddir}` varies depending on how the project was built.".

File: docs/BUILDING.md [hunk 3]
```diff
@@ -275,6 +275,29 @@ export ANDROID_NDK_HOME=~/Library/Android/sdk/ndk/21.0.6113669
...
+#### How to install doxygen on Linux
+
+On Debian-based Linux distributions such as Ubuntu, doxygen can be installed
+with the following:
+
+```
+sudo apt-get install doxygen
+```
+
+#### How to install doxygen on macOS
```
Comment (gerickson) [inline docs/BUILDING.md:293]: doxygen -> Doxygen

File: docs/BUILDING.md [hunk 4]
```diff
@@ -275,6 +275,29 @@ export ANDROID_NDK_HOME=~/Library/Android/sdk/ndk/21.0.6113669
...
+
+On macOS, doxygen can be installed and satisfied using [Brew](https://brew.sh/):
```
Comment (gerickson) [inline docs/BUILDING.md:295]: doxygen -> Doxygen

**Review by gerickson — CHANGES_REQUESTED** [review]

File: docs/BUILDING.md (hunk 2 above)
Comment (vivien-apple) [reply docs/BUILDING.md]: Done.

**Review by gerickson — APPROVED** [review]

**Review by BroderickCarlin — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
@vivien-apple restyle :)

**Comment by vivien-apple** [comment]
I think the restyle job failed in some ways:
stdoutCloning repository
stderrFrom https://github.com/project-chip/connectedhomeip
stderr * [new ref]         refs/pull/1496/head -> pull-1496
stderrSwitched to branch 'pull-1496'
stdoutNo existing Restyled PR
stderrSwitched to a new branch 'restyled/pull-1496'
stdoutRestyling project-chip/connectedhomeip#1496
stdoutRestyling ["docs/BUILDING.md"] via "prettier-markdown"
stderrWARNING: Your kernel does not support swap limit capabilities or the cgroup is not mounted. Memory limited without swap.
stdoutdocs/BUILDING.md 247ms
stdout[restyled/pull-1496 3b4b689] Restyled by prettier-markdown
stdout 1 file changed, 3 insertions(+), 2 deletions(-)
stderrTo https://github.com/project-chip/connectedhomeip.git
stdoutSetting status of error for 9402bb0
stderr ! [remote rejected] restyled/pull-1496 -> restyled/pull-1496 (refusing to allow a GitHub App to create or update workflow `.github/workflows/build.yaml` without `workflows` permission)
stderrerror: failed to push some refs to 'https://<SCRUBBED>@github.com/project-chip/connectedhomeip.git'
stderrWe had trouble running a system command:
stderr
stderr  callProcess: git "push" "--force-with-lease" "origin" "restyled/pull-1496" (exit
stderr  1): failed

**Comment by woody-apple** [comment]
> I think the restyle job failed in some ways:
> stdoutCloning repository
> stderrFrom https://github.com/project-chip/connectedhomeip
> stderr * [new ref] refs/pull/1496/head -> pull-1496
> stderrSwitched to branch 'pull-1496'
> stdoutNo existing Restyled PR
> stderrSwitched to a new branch 'restyled/pull-1496'
> stdoutRestyling #1496
> stdoutRestyling ["docs/BUILDING.md"] via "prettier-markdown"
> stderrWARNING: Your kernel does not support swap limit capabilities or the cgroup is not mounted. Memory limited without swap.
> stdoutdocs/BUILDING.md 247ms
> stdout[restyled/pull-1496 [3b4b689](https://github.com/project-chip/connectedhomeip/commit/3b4b689f260cdc0e16fc964d3daec685453b0333)] Restyled by prettier-markdown
> stdout 1 file changed, 3 insertions(+), 2 deletions(-)
> stderrTo https://github.com/project-chip/connectedhomeip.git
> stdoutSetting status of error for [9402bb0](https://github.com/project-chip/connectedhomeip/commit/9402bb0ad82273f52ad2aee5ce843d15607804db)
> stderr ! [remote rejected] restyled/pull-1496 -> restyled/pull-1496 (refusing to allow a GitHub App to create or update workflow `.github/workflows/build.yaml` without `workflows` permission)
> stderrerror: failed to push some refs to 'https://@github.com/project-chip/connectedhomeip.git'
> stderrWe had trouble running a system command:
> stderr
> stderr callProcess: git "push" "--force-with-lease" "origin" "restyled/pull-1496" (exit
> stderr 1): failed

Ugh, I'll take a look, thanks

**Comment by vivien-apple** [comment]
Is there a command to just relaunch this job ?

**Comment by woody-apple** [comment]
> Is there a command to just relaunch this job ?

I think I need to fix ignore lists for restyled, I'll work on that this afternoon, sorry :(

**Pull request merged**